Source/GameEngineVisuals/Program.cs

```diff
@@ -1,15 +1,590 @@
-﻿using System;
+﻿
+
+
+
+using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace GameEngineVisuals
 {
     class Program
     {
+        const int MAINGRID_X = 32;
+        const int MAINGRID_Y = 16;
+        static char[,] g_cmMainGrid = new char[MAINGRID_X, MAINGRID_Y];
+        private static Mutex g_mutex = new Mutex();
+
+
+
+        public class MapVertex
+        {
+            public MapVertex(int iX, int iY)
+            {
+                m_iX = iX;
+                m_iY = iY;
+            }
+            public int m_iX;
+            public int m_iY;
+        }
+
+
+        static List<MapVertex> g_vPosition = new List<MapVertex>();                   // TODO: currently there's a bug where the last cordinate/index wont get added.
+
+  
+        //-----------------------------------------------------------------------------
+        // loadMapFromFile
+        //-----------------------------------------------------------------------------		
+        // start cordinate index from grid mid to allow for subtraction. ie (iLenght * 0,5).
+        public static void loadMapFromFile(MapVertex mapVertex)
+        {
+
+            string sContent = File.ReadAllText("Map.bsp"); ;
+            char[] cContent = sContent.ToCharArray();
+            int iOffset = 0;
+            //int iPath;
+
+            int i = 0; // mapVertex.m_iX / 2;   // 5, start at mid
+            int j = mapVertex.m_iY / 2;   // 5, start at mid
+
+            //int i = 5;   // 5, start at mid
+            //int j = 5;   // 5, start at mid
+
+            for (int x = 0; x != sContent.Length; x++)
+            {
+
+                switch (cContent[x])
+                {
+
+                    case 'D':
+                        //iPath = direction.down;
+
+                        for (int g = 0; g != iOffset; g++)
+                        {
+                            MapVertex vertex = new MapVertex(i, j);
+                            g_vPosition.Add(vertex);
```

Comment:
Naming: A bit hard to understand `g_vPosition` by looking at its name, consider giving it a more descriptive name.